src/go/rpk/pkg/cli/cmd/start_test.go

```diff
@@ -148,3 +159,578 @@ func TestParseSeeds(t *testing.T) {
+			mgr := config.NewManager(fs)
+			conf, err := mgr.Read(config.Default().ConfigFile)
+			require.NoError(st, err)
+			expectedAddr := config.SocketAddress{
+				Address: "192.168.34.32",
+				Port:    33145,
+			}
+			// Check that the generated config is as expected.
+			require.Exactly(
+				st,
+				expectedAddr,
+				conf.Redpanda.RPCServer,
+			)
+		},
+	}, {
+		name: "it should fail if --rpc-addr is sinvalid",
+		args: []string{
+			"--install-dir", "/var/lib/redpanda",
+			"--rpc-addr", "host:nonnumericport",
+		},
+		expectedErrMsg: "Port must be an int",
+	}, {
+		name: "it should fall back to REDPANDA_RPC_ADDRESS",
+		args: []string{
+			"--install-dir", "/var/lib/redpanda",
+		},
+		before: func(_ afero.Fs) error {
+			os.Setenv("REDPANDA_RPC_ADDRESS", "host:3123")
+			return nil
+		},
+		after: func() {
+			os.Unsetenv("REDPANDA_RPC_ADDRESS")
+		},
+		postCheck: func(fs afero.Fs, st *testing.T) {
+			mgr := config.NewManager(fs)
+			conf, err := mgr.Read(config.Default().ConfigFile)
+			require.NoError(st, err)
+			expectedAddr := config.SocketAddress{
+				Address: "host",
+				Port:    3123,
+			}
+			// Check that the generated config is as expected.
+			require.Exactly(
+				st,
+				expectedAddr,
+				conf.Redpanda.RPCServer,
+			)
+		},
+	}, {
+		name: "it should leave the RPC addr untouched if the env var & flag weren't set",
+		args: []string{
+			"--install-dir", "/var/lib/redpanda",
+		},
+		before: func(fs afero.Fs) error {
+			mgr := config.NewManager(fs)
+			conf := config.Default()
+			conf.Redpanda.RPCServer = config.SocketAddress{
+				Address: "192.168.33.33",
+				Port:    9892,
+			}
+			return mgr.Write(conf)
+		},
+		postCheck: func(fs afero.Fs, st *testing.T) {
+			mgr := config.NewManager(fs)
+			conf, err := mgr.Read(config.Default().ConfigFile)
+			require.NoError(st, err)
+			expectedAddr := config.SocketAddress{
+				Address: "192.168.33.33",
+				Port:    9892,
+			}
+			// Check that the generated config is as expected.
+			require.Exactly(
+				st,
+				expectedAddr,
+				conf.Redpanda.RPCServer,
+			)
+		},
+	}, {
+		name: "it should parse the --kafka-addr and persist it",
+		args: []string{
+			"--install-dir", "/var/lib/redpanda",
+			"--kafka-addr", "192.168.34.32:33145",
+		},
+		postCheck: func(fs afero.Fs, st *testing.T) {
+			mgr := config.NewManager(fs)
+			conf, err := mgr.Read(config.Default().ConfigFile)
+			require.NoError(st, err)
+			expectedAddr := config.SocketAddress{
+				Address: "192.168.34.32",
+				Port:    33145,
+			}
+			// Check that the generated config is as expected.
+			require.Exactly(
+				st,
+				expectedAddr,
+				conf.Redpanda.KafkaApi,
+			)
+		},
+	}, {
+		name: "it should fail if --kafka-addr is sinvalid",
+		args: []string{
+			"--install-dir", "/var/lib/redpanda",
+			"--kafka-addr", "host:nonnumericport",
+		},
+		expectedErrMsg: "Port must be an int",
+	}, {
+		name: "it should fall back to REDPANDA_KAFKA_ADDRESS",
```

Comment:
Does this mean the env should be persisted? Fall back doesn't sound like affirmative action.

"it should parse REDPANDA_KAFKA_ADDRESS and persist it" sounds better. There are more than one of these.

Reply:
Sounds good to me! The thing is, it should "parse X and persist it" if its corresponding flag wasn't set.